gcloud-java-dns/src/main/java/com/google/gcloud/dns/Dns.java

```diff
@@ -0,0 +1,537 @@
+      fieldStrings.add(ID.selector());
+      for (ChangeRequestField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The sorting order for listing.
+   */
+  enum SortingOrder {
+    DESCENDING, ASCENDING;
+
+    public String selector() {
+      return name().toLowerCase();
+    }
+  }
+
+  /**
+   * Class for specifying record set listing options.
+   */
+  class RecordSetListOption extends AbstractOption implements Serializable {
+
+    private static final long serialVersionUID = 1009627025381096098L;
+
+    RecordSetListOption(DnsRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the record set's fields to be returned by the RPC call.
+     *
+     * <p>If this option is not provided all record fields are returned. {@code
+     * RecordSetField.fields} can be used to specify only the fields of interest. The name of the
+     * record set in always returned, even if not specified. {@link RecordSetField} provides a list
+     * of fields that can be used.
+     */
+    public static RecordSetListOption fields(RecordSetField... fields) {
+      StringBuilder builder = new StringBuilder();
+      builder.append("nextPageToken,rrsets(").append(RecordSetField.selector(fields))
+          .append(')');
+      return new RecordSetListOption(DnsRpc.Option.FIELDS, builder.toString());
+    }
+
+    /**
+     * Returns an option to specify a page token.
+     *
+     * <p>The page token (returned from a previous call to list) indicates from where listing should
+     * continue.
+     */
+    public static RecordSetListOption pageToken(String pageToken) {
+      return new RecordSetListOption(DnsRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * The maximum number of record sets to return per RPC.
+     *
+     * <p>The server can return fewer record sets than requested. When there are more results than
+     * the page size, the server will return a page token that can be used to fetch other results.
+     */
+    public static RecordSetListOption pageSize(int pageSize) {
+      return new RecordSetListOption(DnsRpc.Option.PAGE_SIZE, pageSize);
+    }
+
+    /**
+     * Restricts the list to only record sets with this fully qualified domain name.
+     */
+    public static RecordSetListOption dnsName(String dnsName) {
+      return new RecordSetListOption(DnsRpc.Option.NAME, dnsName);
+    }
+
+    /**
+     * Restricts the list to return only record sets of this type. If present, {@link
+     * RecordSetListOption#dnsName(String)} must also be present.
+     */
+    public static RecordSetListOption type(RecordSet.Type type) {
+      return new RecordSetListOption(DnsRpc.Option.DNS_TYPE, type.name());
+    }
+  }
+
+  /**
+   * Class for specifying zone field options.
+   */
+  class ZoneOption extends AbstractOption implements Serializable {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://www.codacy.com/app/mziccard/gcloud-java/file/2317493773/issues/source?bid=3123109&fileBranchId=3203698#l247)